ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -0,0 +1,884 @@
+   * that are still in their temporary uncommitted state.
+   * @return the list of target log segments that exist on the disk.
+   */
+  private List<String> getTargetLogSegmentNames() {
+    List<String> filenames = Arrays.asList(dataDir.list(TEMP_LOG_SEGMENTS_FILTER));
+    List<String> names = new ArrayList<>(filenames.size());
+    for (String filename : filenames) {
+      String transformed = filename.substring(0, filename.length() - TEMP_LOG_SEGMENT_NAME_SUFFIX.length());
+      names.add(LogSegmentNameHelper.nameFromFilename(transformed));
+    }
+    Collections.sort(names, LogSegmentNameHelper.COMPARATOR);
+    return names;
+  }
+
+  /**
+   * Renames the temporary log segment files to remove the temporary file name suffix and effectively commits them on
+   * disk.
+   * @param logSegmentNames the names of the log segments whose backing files need to be renamed.
+   * @throws StoreException if the rename fails.
+   */
+  private void renameLogSegments(List<String> logSegmentNames) throws StoreException {
+    for (String segmentName : logSegmentNames) {
+      String filename = LogSegmentNameHelper.nameToFilename(segmentName);
+      File from = new File(dataDir, filename + TEMP_LOG_SEGMENT_NAME_SUFFIX);
+      File to = new File(dataDir, filename);
+      if (!from.renameTo(to)) {
+        // TODO: change error code
+        throw new StoreException("Failed to rename " + from + " to " + to, StoreErrorCodes.Unknown_Error);
+      }
+    }
+  }
+
+  /**
+   * Commits the log segments with names {@code logSegmentNames}. Committing involves adding them to the application log
+   * instance.
+   * @param logSegmentNames the names of the log segments to commit.
+   * @param recovering {@code true} if this function was called in the context of recovery. {@code false} otherwise.
+   * @throws IOException if there were any I/O errors creating a log segment
+   */
+  private void commitLogSegments(List<String> logSegmentNames, boolean recovering) throws IOException {
+    for (String logSegmentName : logSegmentNames) {
+      File segmentFile = new File(dataDir, LogSegmentNameHelper.nameToFilename(logSegmentName));
+      LogSegment segment = new LogSegment(logSegmentName, segmentFile, srcMetrics);
+      srcLog.addSegment(segment, recovering);
+    }
+  }
+
+  /**
+   * Commits all the index segments that refer to log segments in {@code logSegmentNames}. The commit is atomic with
+   * the use of {@link PersistentIndex#changeIndexSegments(List, Set)}.
+   * @param logSegmentNames the names of the log segments whose index segments need to be committed.
+   * @throws IOException if there were any I/O errors in setting up the index segment.
+   * @throws StoreException if there were any problems committing the changed index segments.
+   */
+  private void commitIndexSegments(List<String> logSegmentNames) throws IOException, StoreException {
```

Comment:
Same here - the index segments were already created with the right name and renaming the log segments "committed" them.